benchadapt/python/benchadapt/adapters/_adapter.py

```diff
@@ -57,13 +76,46 @@ def run(self, params: List[str] = None) -> List[BenchmarkResult]:
 
         return self.results
 
-    @abc.abstractmethod
     def transform_results(self) -> List[BenchmarkResult]:
         """
         Method to transform results from the command line call into a list of
-        instances of `BenchmarkResult`.
+        instances of `BenchmarkResult`. This method returns results updated
+        with runtime metadata values specified on init.
+        """
+        results = self._transform_results()
+        self.results = [self.update_benchmark_result(res) for res in results]
+        return self.results
+
+    @abc.abstractmethod
+    def _transform_results(self) -> List[BenchmarkResult]:
+        """
+        Method to transform results from the command line call into a list of
+        instances of `BenchmarkResult`. The results of this method will be
+        updated to apply runtime metadata values specified on init.
         """
 
+    def update_benchmark_result(self, result: BenchmarkResult) -> BenchmarkResult:
+        """
+        A method to update instances of `BenchmarkResult` with vallues specified on
```

Comment:
```suggestion
        A method to update instances of `BenchmarkResult` with values specified on
```

Reply:
Oops I should have refreshed before posting my review 😄